src/engine.rs

```diff
@@ -170,6 +168,37 @@ pub(crate) async fn start(
                                     // TryFrom trait for Vec<u8> to Message, implement a function that takes in the Message Topic to help
                                     // converting Vec<u8> to Message. You can refer to fullnode/mempool messagegate to see how to 
                                     // deserialise each Message type. 
+                                    match message.topic.as_str() {
```

Comment:
I understand you couldn't do branches like Topic::Mempool.hash() in match statement. But comparing the topic using string doesn't seem a good idea. 

 `let topic = fullnode_topics(public_addr).into_iter().find(|t| t.hash() == message.topic);`

Will this way of identifying the topic work? I haven't verified, you can try.